client/src/controllers/KeyboardController.test.js

```diff
@@ -0,0 +1,67 @@
+import { Application } from '@hotwired/stimulus';
+import { KeyboardController } from './KeyboardController';
+
+describe('KeyboardController', () => {
+  let app;
+  const oldWindowLocation = window.location;
+
+  const setup = async (html) => {
+    document.body.innerHTML = `<main>${html}</main>`;
+
+    app = Application.start();
+    app.register('w-kbd', KeyboardController);
+
+    await Promise.resolve();
+  };
+
+  beforeAll(() => {
+    delete window.location;
+
+    window.location = Object.defineProperties(
+      {},
+      {
+        ...Object.getOwnPropertyDescriptors(oldWindowLocation),
+        assign: { configurable: true, value: jest.fn() },
+      },
+    );
+  });
```

Comment:
Alongside the changes above, replace this beforeAll callback with the setup of the prototype click override - this will run for the whole test this way.

```suggestion
  beforeAll(() => {
    HTMLButtonElement.prototype.click = buttonClickMock;
  });
```